src/common/Widget.js

```diff
@@ -578,6 +579,29 @@
     };
 
     //  DOM/SVG Node Helpers  ---
+    
+    Widget.prototype.showBusy = function (show) {
+        if (show) {
+            //show the indicator
+        } else {
+            //hide the indicator
+        }
+    };
+
+    Widget.prototype.pushBusy = function () {
+        this._busyCounter++;
+        if (this._busyCounter > 0) {
+            this.showBusy(true);
+        }
+    };
+
+    Widget.prototype.popBusy = function () {
+        this._busyCounter--;
+        if (this._busyCounter <= 0) {
```

Comment:
```this._busyCounter === 0``` will keep the showBusy code simpler which is good as it may be overriden by someone